dbms/src/Storages/MergeTree/PKCondition.cpp

```diff
@@ -1005,6 +1005,8 @@ bool PKCondition::mayBeTrueInRangeImpl(const std::vector<Range> & key_ranges, co
                 {
                     rpn_stack.back() = !rpn_stack.back();
                 }
+            } else {
+                rpn_stack.emplace_back(true, true);
```

Comment:
This will lead to full scan for predicates like `key_column IN (something)` if the corresponding prepared set is not filled for some reason. Better throw an exception.